python/tests/test_aac/in_out/test_parser.py

```diff
@@ -143,124 +145,154 @@ def test_does_not_add_doc_separator_if_not_already_present(self):
+        with TemporaryDirectory() as temp_dir:
+            initial_file = self.create_test_file(f"{temp_dir}/{TEST_MESSAGE_FILE_NAME}", test_message_contents_with_bad_import)
+            empty_file = self.create_test_file(f"{temp_dir}/{empty_file_name}", "")
+            
+            definitions = parse(initial_file)
+            import_definitions = [definition for definition in definitions if definition.get_root_key() == "import"]
+
+            self.assertTrue(isfile(empty_file))
 
             """ Asserting that the bad import definition was parsed and that this assertion was allowed to complete without error is the
                 assertion that the parse mechanism softly handled the bad import. """
-            self.assertEqual(non_existent_import_definition, parsed_non_existent_import_definition)
+            self.assertEqual(len(import_definitions), 1)
 
     def test_parsed_file_imports_invalid_file(self):
         invalid_yaml_file_name = "invalid_yaml_file.aac"
-        non_existent_import_definition = create_import_definition([invalid_yaml_file_name])
-        test_message_contents_with_bad_import = YAML_DOCUMENT_SEPARATOR.join([non_existent_import_definition.to_yaml(), TEST_MESSAGE_CONTENTS])
-        with (
-            TemporaryDirectory() as temp_dir,
-            TemporaryAaCTestFile(test_message_contents_with_bad_import, dir=temp_dir, name=TEST_MESSAGE_FILE_NAME) as initial_file,
-            TemporaryAaCTestFile("", dir=temp_dir, name=invalid_yaml_file_name) as invalid_yaml_file,
-        ):
-            definitions = parse(initial_file.name)
-            parsed_non_existent_import_definition, *_ = collections.get_definitions_by_root_key(ROOT_KEY_IMPORT, definitions)
-
-            self.assertTrue(isfile(invalid_yaml_file.name))
+        imvalid_yaml_file_content = f"""
+import:
+  files:
+    - {invalid_yaml_file_name}
+"""
+        test_message_contents_with_bad_import = YAML_DOCUMENT_SEPARATOR.join([imvalid_yaml_file_content, TEST_MESSAGE_CONTENTS])
+        with TemporaryDirectory() as temp_dir:
+            initial_file = self.create_test_file(f"{temp_dir}/{TEST_MESSAGE_FILE_NAME}", test_message_contents_with_bad_import)
+            empty_file = self.create_test_file(f"{temp_dir}/{invalid_yaml_file_name}", "")
+            definitions = parse(initial_file)
+            import_definitions = [definition for definition in definitions if definition.get_root_key() == "import"]
+
+            self.assertTrue(isfile(empty_file))
 
             """ Asserting that the bad import definition was parsed and that this assertion was allowed to complete without error is the
                 assertion that the parse mechanism softly handled the bad import. """
-            self.assertEqual(non_existent_import_definition, parsed_non_existent_import_definition)
+            self.assertEqual(len(import_definitions), 1)
 
 
 TEST_MODEL_NAME = "EchoService"
```

Comment:
Should these model/schema strings be moved to their own file?

Reply:
Need a little more info to understand the comment.